### DISCUSSION TITLE
Preview Images Shrinking

### DISCUSSION THREAD

**Discussion by pseudology** [cotes2020/jekyll-theme-chirpy#1317] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

Everything was working good, but suddenly my preview images have started shrinking in my blogs homepage.
Anyone have a solution or face a similar issue?
Also when i run the build on my local host the images dont get cropped, but when i run it on github oages and publish it to my domain, i start seeing the issue.

### Operations you have already tried

I have tried changing images, using better cropped images, but everything seems to be fine on my localhost.

**Comment by joshjohanning** [comment]
Shrinking, meaning some of the image gets cropped?  I notice something, and it really depends on what the width of your browser is.  Seems to always look the best on mobile.

**Comment by tristan-white** [comment]
I'm also having this issue, but not sure why. I tried to dig a little, but have not front end experience and decided to look for answers in this repo.

Specifically, my issue is that occasionally images in my posts to not render correctly when clicking the preview image button (ie clicking an image to trigger an event to have it fill the whole screen). 

On desktop, when clicking an image, the image only fills part of the preview pane; the rest is whitespace.
On mobile, the image will be too zoomed in.

For examples, see [this page](https://tristanwhite.me/posts/azores/). The raw markdown (with bits of html mixed in) is [here](https://raw.githubusercontent.com/tristan-white/tristan-white.github.io/refs/heads/main/_posts/2024-11-19-azores.md). Note that the first two pictures (of hotel rooms) render correctly when previewing, while the next image (a picture of a breakfast) does not.

**Reply by SoumyaK4** [reply]
@tristan-white the breakfast image and other images with errors are actually rendering the whole image
But it's zoomed in for some reason, if you scroll to left or bottom you'll see.

The reason likely the following :
![Screenshot_20241209-003919](https://github.com/user-attachments/assets/3c73c164-3a8a-4599-985f-e1fac8a0d5c8)

The src url ends with w=2400, which probably makes the images look like that.
Unless you have a huge monitor it will always look zoomed in.

P.S. I'd recommend uploading the files inside the repo and using relative path for src. It keeps the code clean and easy to read.

**Reply by tristan-white** [reply]
Thanks for the reply - I switched from using hotlinks to local photos and this fixed the problem. Still curious if anyone know the answer: why would switching to local photos fix the problem if the hotlinked photo and the local photo are the same size?

**Reply by SoumyaK4** [reply]
@tristan-white  
> 
> The reason : ![Screenshot_20241209-003919](https://private-user-images.githubusercontent.com/26403943/393635683-3c73c164-3a8a-4599-985f-e1fac8a0d5c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM2ODkyOTksIm5iZiI6MTczMzY4ODk5OSwicGF0aCI6Ii8yNjQwMzk0My8zOTM2MzU2ODMtM2M3M2MxNjQtM2E4YS00NTk5LTk4NWYtZTFmYWM4YTBkNWM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA4VDIwMTYzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYwYjhmYWYxZTUxOTViNWE2NDIzZTExYzU1MDk4Y2NjOTZjMTZhYWM4MDEyZTQzNTY2ZTNjMGRjYjhkZTkzMzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.p2E9hqFygTGHyKmjE-CNZFv22hwUmpmnidbV8KkX314)
> 
> The src url ends with w2400, which sets the image width to 2400 hence the problem. Unless you have a huge monitor it will always look zoomed in.
> 

I just mentioned it above

**Reply by tristan-white** [reply]
Sorry thank you yes forgot to mention I tried changing that width to other widths such as 700 and nothing changed. Issues is resolved now though so not a big deal
